src/kroneckerpowers.jl

```diff
@@ -118,8 +120,10 @@ function Base.conj(K::KroneckerPower)
 end
 
 # mixed-product property
-function Base.:*(K1::KroneckerPower{T1, N},
-                        K2::KroneckerPower{T2, N}) where {T1, T2, N}
+function Base.:*(
+    K1::KroneckerPower{T1, TA1, N},
+    K2::KroneckerPower{T1, TA2, N},
```

Comment:
Why do you require that both Kronecker factors have the same `eltype` `T1`? I don't think that's necessary.